lms/djangoapps/courseware/tests/test_model_data.py

```diff
@@ -230,15 +242,18 @@ def test_get_field_from_missing_student_module(self):
 
     def test_set_field_in_missing_student_module(self):
         "Test that setting a field in a missing StudentModule creates the student module"
-        self.assertEquals(0, len(self.field_data_cache.cache))
+        self.assertEquals(0, sum(len(cache) for cache in self.field_data_cache.cache.values()))
```

Comment:
This could have been a good place for a __len__ method on self.field_data_cache.cache